pipeline-scripts/buildlib.groovy

```diff
@@ -437,6 +452,105 @@ invalid mode build != master and no release branch
     return mode
 }
 
+//
+// Create a new version string based on the build mode
+//
+//@NonCPS
+def new_version(mode, version_string, release_string) {
+
```

Comment:
Can you document some valid input/outputs with those fancy `@param` comments you use so well

Reply:
You silver-tongued dude you.  Yep will do.